test/rules/no_leading_underscores_for_local_identifiers_test.dart

```diff
@@ -129,6 +141,20 @@ f() {
     ]);
   }
 
+  test_objectPattern_switch_field() async {
+    await assertNoDiagnostics(r'''
+class A {
+  var _a;
+}
+
+f(A a) {
+  switch (a) {
+    case A(:var _a):
```

Comment:
Also consider testing:
```dart
case A(:var _a?):
case A(:var _a!):
```
(IMHO the lint should also be suppressed in these cases)

Reply:
Ah, great catch. I'd have thought these case would "just work" but I need to handle `NullCheckPattern`s and `NullAssertPattern`s explicitly. Thank you!